src/pyggdrasil/scripts/cell_simulation.py

```diff
@@ -0,0 +1,398 @@
+        None
+    """
+    graph = nx.from_numpy_array(adj_matrix, create_using=nx.DiGraph)
+    print(nx.forest_str(graph, sources=[root]))
+
+
+def compose_save_name(params: dict, *, tree_no: int) -> str:
+    """Composes save name for the results."""
+    save_name = (
+        f"seed_{params['seed']}_"
+        f"n_trees_{params['n_trees']}_"
+        f"n_cells_{params['n_cells']}_"
+        f"n_mutations_{params['n_mutations']}_"
+        f"alpha_{params['alpha']}_"
+        f"beta_{params['beta']}_"
+        f"na_rate_{params['na_rate']}_"
+        f"observe_homozygous_{params['observe_homozygous']}_"
+        f"strategy_{params['strategy']}"
+    )
+    if tree_no is not None:
+        save_name += f"_tree_{tree_no}"
+    return save_name
+
+
+def adjacency_to_root_dfs(adj_matrix: np.ndarray) -> TreeNode:
+    """Convert adjacency matrix to tree in tree.TreeNode
+        traverses a tree using depth first search.
+
+    Args:
+        adj_matrix: np.ndarray
+            with no self-loops (i.e. diagonal is all zeros)
+            and the root as the highest index node
+    Returns:
+        root: TreeNode containing the entire tree
+    """
+
+    # Determine the root node (node with highest index)
+    root_idx = len(adj_matrix) - 1
+
+    # Create a stack to keep track of nodes to visit
+    stack = [root_idx]
+
+    # Create a set to keep track of visited nodes
+    visited = set()
+
+    # Create a list to keep track of nodes
+    child_parent = {}
+    child_parent[root_idx] = None
+
+    # Create a list to keep track of TreeNodes
+    list_TreeNode = [np.nan] * len(adj_matrix)
+
+    # Traverse the tree using DFS
+    while stack:
+        # Get the next node to visit
+        node = stack.pop()
+
+        # Skip if already visited
+        if node in visited:
+            # print(f"Already Visited node {node}")
+            continue
+
+        # Visit the node
+        # print(f"Visiting node {node}")
+
+        # Recall parent
+        parent = child_parent[node]
+        # print(f"Parent of node {node} is {child_parent[node]}")
+
+        if node == root_idx:
+            root = TreeNode(name=node, data=None, parent=None)
+            list_TreeNode[node] = root
+        else:
+            child = TreeNode(name=node, data=None, parent=list_TreeNode[parent])
+            list_TreeNode[node] = child
+
+        # Add to visited set
+        visited.add(node)
+
+        # Add children to the stack
+        # (in reverse order to preserve order in adjacency matrix)
+        for child in reversed(range(len(adj_matrix))):
+            if adj_matrix[node][child] == 1 and child not in visited:
+                stack.append(child)
+                # print(f"Adding node {child} to stack")
+                # Commit Parent to Memory
+                child_parent[child] = node
+
+    root = list_TreeNode[root_idx]
+    return root
+
+
+def dummy_serialize(root: TreeNode):
+    """Dummy function to serialize the data a TreeNode."""
+    pass
```

Comment:
This is one way of doing it. As an alternative you can just pass `lambda x: None` and it will work as well :slightly_smiling_face: 

(I was considering making it the default behavior. However, then there's a risk that some data which took hours to generate wouldn't be saved because someone forgot to specify the serialization function here. So passing `lambda x: None` is currently the recommended way of dong it.)